offchain/eth-input-reader/src/eth_input_reader.rs

```diff
@@ -75,8 +97,8 @@ pub async fn start(
                     b.parent_hash
                 );
                 process_block(
+                    &Arc::clone(&env),
```

Comment:
This line looks weird to me. Questions:

Why is `env` wrapped in `Arc`? Why cloning it every loop iteration? If it's a reference, why clone before taking the reference?